src/nemos/basis/basis.py

```diff
@@ -0,0 +1,1478 @@
+        n_basis_funcs: int,
+        window_size: int,
+        order: int = 4,
+        label: Optional[str] = "MSplineConv",
+        conv_kwargs: Optional[dict] = None,
+    ):
+        ConvBasisMixin.__init__(self, window_size=window_size, conv_kwargs=conv_kwargs)
+        MSplineBasis.__init__(
+            self,
+            n_basis_funcs,
+            mode="conv",
+            order=order,
+            label=label,
+        )
+
+    @add_docstring("split_by_feature", MSplineBasis)
+    def split_by_feature(
+        self,
+        x: NDArray,
+        axis: int = 1,
+    ):
+        r"""
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from nemos.basis import MSplineConv
+        >>> from nemos.glm import GLM
+        >>> basis = MSplineConv(n_basis_funcs=6, window_size=10, label="two_inputs")
+        >>> X_multi = basis.compute_features(np.random.randn(20, 2))
+        >>> split_features_multi = basis.split_by_feature(X_multi, axis=1)
+        >>> for feature, sub_dict in split_features_multi.items():
+        ...        print(f"{feature}, shape {sub_dict.shape}")
+        two_inputs, shape (20, 2, 6)
+
+        """
+        return super().split_by_feature(x, axis=axis)
+
+    @add_docstring("_compute_features", ConvBasisMixin)
+    def compute_features(self, xi: ArrayLike) -> FeatureMatrix:
+        """
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from nemos.basis import MSplineConv
+
+        >>> # Generate data
+        >>> num_samples = 1000
+        >>> X = np.random.normal(size=(num_samples, ))  # raw time series
+        >>> basis = MSplineConv(10, window_size=11)
+        >>> features = basis.compute_features(X)  # basis transformed time series
+        >>> features.shape
+        (1000, 10)
+
+        """
+        return super().compute_features(xi)
+
+    @add_docstring("evaluate_on_grid", MSplineBasis)
+    def evaluate_on_grid(self, n_samples: int) -> Tuple[NDArray, NDArray]:
+        """
+        Examples
+        --------
+        Evaluate and visualize 4 M-spline basis functions of order 3:
+
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+        >>> from nemos.basis import MSplineConv
+        >>> mspline_basis = MSplineConv(n_basis_funcs=4, order=3, window_size=10)
+        >>> sample_points, basis_values = mspline_basis.evaluate_on_grid(100)
+        >>> for i in range(4):
+        ...     p = plt.plot(sample_points, basis_values[:, i], label=f'Function {i+1}')
+        >>> plt.title('M-Spline Basis Functions')
+        Text(0.5, 1.0, 'M-Spline Basis Functions')
+        >>> plt.xlabel('Domain')
+        Text(0.5, 0, 'Domain')
+        >>> plt.ylabel('Basis Function Value')
+        Text(0, 0.5, 'Basis Function Value')
+        >>> l = plt.legend()
+        """
+        return super().evaluate_on_grid(n_samples)
+
+
+class RaisedCosineLinearEval(
+    EvalBasisMixin, RaisedCosineBasisLinear, BasisTransformerMixin
```

Comment:
why is BasisTransformerMixin inherited here? doesn't it get it through RaisedCosineBasisLinear (which gets it from Basis)?